packages/mobile/src/tokens/selectors.ts

```diff
@@ -60,3 +61,23 @@ export const defaultTokenSelector = createSelector(tokensListSelector, (tokens)
 
   return maxTokenAddress
 })
+
+export const totalTokenBalanceSelector = createSelector(
+  [tokensWithBalanceSelector, localCurrencyExchangeRatesSelector],
+  (tokenBalances, exchangeRate) => {
+    const usdRate = exchangeRate[Currency.Dollar]
+    if (!usdRate) {
+      return null
+    }
+    let totalBalance = new BigNumber(0)
+
+    for (const token of tokenBalances) {
+      const tokenAmount = new BigNumber(token.balance)
+        .multipliedBy(token.usdPrice)
+        .multipliedBy(usdRate)
+      totalBalance = totalBalance.plus(tokenAmount)
+    }
+
+    return totalBalance.toFixed(2).toString()
```

Comment:
I can't think of any case where we might want to use the total as a `BigNumber` instead of a string, but I think exposing this as a `BigNumber` gives us more flexibility in the future in case such a use case does pop up. This is an optional comment though.